### PR TITLE
[batch-beta-dev] Feature/batch sample documentation

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
The content to be added to Changelog is as follows:
* Additional updates to sample documentation

### CODE REVIEW

File: Documentation/BatchDocumentation/samples/mpi/readme.md [hunk 1]
```diff
@@ -5,35 +5,81 @@ This template shows how to use `MS-MPI` to run MPI work.
...
-Run command `azure batch file upload <path> <group>` on a folder containing MPIHelloWorld.exe and its dependencies which are named with numerically increasing names with `sample` prefix (i.e. `sample1.wav`, `sample2.wave`, `sample3.wav`, etc).
 
-**Build a Release version of MPIHelloWorld.exe so that you don't have to include any additional dependencies as resource files (ex: msvcp140d.dll or vcruntime140d.dll).
+Upload files from a folder:
+
+```bash
+azure batch file upload <path> <group>
+```
+
+Run this in a folder containing MPIHelloWorld.exe and its dependencies which are named with numerically increasing names with `sample` prefix (i.e. `sample1.wav`, `sample2.wave`, `sample3.wav`, etc).
```
Comment (jasper-schneider) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: sample2.wave -> sample2.wav

Comment (jasper-schneider) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: In your other PR, you reworded similar text like this:
The parametric sweep expects the files to be named `sample1.wav`, `sample2.wav`, `sample3.wav` and so on - each with the prefix `sample` and an increasing index number. It's important that your files are sequentially numbered with no gaps.

I like that better than the existing version.

File: Documentation/BatchDocumentation/samples/mpi/readme.md [hunk 2]
```diff
@@ -5,35 +5,81 @@ This template shows how to use `MS-MPI` to run MPI work.
...
+
+If you want to change the default values of the pool creation, you can create a JSON file to supply the parameters of your pool and run:
 
-If you want to change the default values of the pool creation, you can create a JSON file to supply the parameters of your pool, run `azure batch pool create --template pool.json --parameters <your settings JSON file>`.
+```bash
+azure batch pool create --template pool.json --parameters <your settings JSON file>
+```
 
-**You are billed for your Azure Batch pools, so don't forget to delete it when you're done.**
+**You are billed for your Azure Batch pools, so don't forget to delete it through the [Azure portal](https://portal.azure.com) when you're done.**
```
Comment (jasper-schneider) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: This addition is fine, but we also have a command to do this without leaving the CLI session, which may be more convenient.  Offhand, I believe it's:
azure batch pool delete <id> [--quiet]

You don't have to change anything here if you don't want though.

File: Documentation/BatchDocumentation/samples/ocr/readme.md [hunk 3]
```diff
@@ -1,36 +1,78 @@
...
+``` bash
+azure batch task list --job-id <jobid>`
+```
+You can also use the [Azure portal](https://portal.azure.com) or [Batch Explorer](https://github.com/Azure/azure-batch-samples/tree/master/CSharp/BatchExplorer) for monitoring.
+
+### Structure of the sample
+
+| File                  | Content                                                                                                |
+| --------------------- | ------------------------------------------------------------------------------------------------------ |
+| `pool.json`           | A template for creating the pool required for OCR processing                                           |
```
Comment (jasper-schneider) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: minor: Add a period to the end of the description.

File: lib/commands/batch/batch.jobUtils._js [hunk 4]
```diff
@@ -25,12 +25,17 @@ var batchJobUtils = {};
 batchJobUtils.deployTasks = function (jobId, tasks, options, _) {
   var MAX_TASKS_COUNT_IN_BATCH = 100;
   var client = batchUtil.createBatchServiceClient(options);
+    var resultMapper = new client.models['TaskAddParameter']().mapper();
```
Comment (jasper-schneider) [inline lib/commands/batch/batch.jobUtils._js]: I think there's 2 extra spaces here?

File: lib/commands/batch/batch.jobUtils._js [hunk 5]
```diff
@@ -25,12 +25,17 @@ var batchJobUtils = {};
...
   try {
     var start = 0;
     while (start < tasks.length) {
       var end = Math.min(start + MAX_TASKS_COUNT_IN_BATCH, tasks.length);
       var taskList = tasks.slice(start, end);
-      client.task.addCollection(jobId, taskList, _);
+            var addTasks = [];
```
Comment (jasper-schneider) [inline lib/commands/batch/batch.jobUtils._js]: Looks like you're using 4 space indents instead of 2 like the original code here?

Comment (jasper-schneider) [inline lib/commands/batch/batch.jobUtils._js]: Did you actually write this, or is Xing's latest change getting jumbled in with yours somehow?  Just wondering if we missed another instance of this :)

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 2 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: I considered trying to enumerate the different ways of doing this, but decided to stick with one simple way - if the user knows about others, I expect they'll recognize that the portal reference is just a suggestion.

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 1 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: Dang. I thought I'd caught the wave -> wav thing. Fixed, as well as reusing the alternate wording.

File: Documentation/BatchDocumentation/samples/ocr/readme.md (hunk 3 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: Fixed.

File: lib/commands/batch/batch.jobUtils._js (hunk 5 above)
Comment (theunrepentantgeek) [inline lib/commands/batch/batch.jobUtils._js]: Hmmm ... my changes are documentation only, but I started out by merging from `batch-beta-dev` to have a sound baseline. I'll sort this out.

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 2 above)
Comment (jasper-schneider) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: Sure, works for me.

File: lib/commands/batch/batch.jobUtils._js (hunk 5 above)
Comment (theunrepentantgeek) [inline lib/commands/batch/batch.jobUtils._js]: Yeah, I got these changes when I merged from upstream. Not sure why they're showing as diffs given that they are already there.

File: Documentation/BatchDocumentation/samples/blender/readme.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+# Azure Batch Blender using application templates
+
+This sample shows how to use **Blender** to render using a parameterized template that specifies a virtualMachineConfiguration based autopool.
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/blender/readme.md]: The title calls out that this is a 'using application templates' sample whereas the intro paragraph seems to be talking about parameterised templates and IaaS auto pools.

File: Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md [hunk 7]
```diff
@@ -10,11 +10,29 @@ The job template allows you to specify the scope of the parametric sweep through
...
 ## Preparation
-Fill out `pool.parameters.json` and `job.parameters.json`.
+
+Modify the parameters specified in `pool.parameters.json` to configure your pool. Available parameters are defined in `pool.json`:
+
+| Parameter | Required  | Description                                                                                                                 |
+| --------- | --------- | --------------------------------------------------------------------------------------------------------------------------- |
+| poolId    | Mandatory | The id of the Azure Batch pool.                                                                                             |
+| vmCount   | Optional  | The number of virtual machines. <br/> Defaults to **3** if not otherwise specified.                                         |
+| vmSize    | Optional  | The size of the virtual machines that run the application. <br/> Defaults to **STANDARD_D1_V2** if not otherwise specified. |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md]: Why do we randomly use STANDARD_D1 in some places and STANDARD_D1_V2 in others?

File: Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md [hunk 8]
```diff
@@ -10,11 +10,29 @@ The job template allows you to specify the scope of the parametric sweep through
...
+
+
+Modify the parameters specified `job.parameters.json` as appropriate to configure your job. Available parameters are defined in `job.json`:
+
+| Parameter | Required  | Description                                    |
+| --------- | --------- | ---------------------------------------------- |
+| jobId     | Mandatory | The id of Azure Batch job.                     |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md]: *the* Azure Batch job?

File: Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md [hunk 9]
```diff
@@ -10,11 +10,29 @@ The job template allows you to specify the scope of the parametric sweep through
...
+| poolId    | Mandatory | The id of Azure Batch pool which runs the job. |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md]: *the* Azure Batch pool?

File: Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md [hunk 10]
```diff
@@ -10,11 +10,29 @@ The job template allows you to specify the scope of the parametric sweep through
...
+| taskStart | Mandatory | The sweep start parameter.                     |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md]: These names are a bit befuddling and the explanations don't really tell me what their effect is, only how they map to the implementation.  The purpose of a template is to surface things in terms the *user* of the template would understand - by all means use the sample documentation to explain to a template *developer* how the sample works, but parameter docs should be in the domain language.

File: Documentation/BatchDocumentation/samples/mpi/readme.md [hunk 11]
```diff
@@ -5,35 +5,81 @@ This template shows how to use `MS-MPI` to run MPI work.
...
+
+Finally, activate the application package `MPIHelloWorld:1.0`:
+
+```bash
+azure batch application package activate --application-id MPIHelloWorld --version 1.0 --account-name <account name> --resource-group <resource group> --format zip
+```
 
 ## Create a pool
-Run `azure batch pool create --template pool.json` to create your pool using the default settings (A pool named 'MultiInstanceSamplePool' with 3 small VMs). 
+Create your pool using the default settings (A pool named 'MultiInstanceSamplePool' with 3 small VMs):
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: 'A pool' -> 'a' should be lower case.  Is this our normal format for describing pool configurations?

Comment (itowlson) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: I think you have previously been adding blank lines between headings and the next paragraph...?

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 2 above)
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: antecedent of 'it' is 'your Azure Batch pools'

File: Documentation/BatchDocumentation/samples/mpi/readme.md [hunk 12]
```diff
@@ -5,35 +5,81 @@ This template shows how to use `MS-MPI` to run MPI work.
...
-**Build a Release version of MPIHelloWorld.exe so that you don't have to include any additional dependencies as resource files (ex: msvcp140d.dll or vcruntime140d.dll).
+Upload files from a folder:
+
+```bash
+azure batch file upload <path> <group>
+```
+
+Run this in a folder containing MPIHelloWorld.exe and its dependencies. The parametric sweep expects the files to be named `sample1.wav`, `sample2.wav`, `sample3.wav` and so on - each with the prefix `sample` and an increasing index number. It's important that your files are sequentially numbered with no gaps.
+
+**Recommended**: Build a Release version of MPIHelloWorld.exe so that you don't have to include any additional dependencies as resource files (e.g.: `msvcp140d.dll` or `vcruntime140d.dll`).
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: Confused.  Above we told them to upload MPIHelloWorld as an application package and now we are talking about program files as resource files?

File: Documentation/BatchDocumentation/samples/mpi/readme.md [hunk 13]
```diff
@@ -5,35 +5,81 @@ This template shows how to use `MS-MPI` to run MPI work.
...
 
 ## Create a job with MPI task
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: "an MPI task"??

File: Documentation/BatchDocumentation/samples/mpi/readme.md [hunk 14]
```diff
@@ -5,35 +5,81 @@ This template shows how to use `MS-MPI` to run MPI work.
...
-Run `azure batch job create --template job.json` to create your job using the default settings. If you want to configure other options of the job, such as the the pool id, you can look in the `job.json` parameters section to see what options are available.
+
+To create your  job with default settings:
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: extraneous space in Markdown source

File: Documentation/BatchDocumentation/samples/hello-world/taskPerFile-appTemplate/readme.md [hunk 15]
```diff
@@ -51,18 +51,10 @@ You can also use the [Azure portal](https://portal.azure.com) or [Batch Explorer
 
 ## Structure of the Sample
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/hello-world/taskPerFile-appTemplate/readme.md]: "Sample" was not capitalised in previous occurrence of "Structure of the sample" - looks like we are sentence casing most headings, which I think is MSDN house style (Tamra may know for sure)

File: Documentation/BatchDocumentation/samples/ocr/readme.md [hunk 16]
```diff
@@ -1,36 +1,78 @@
 # Azure Batch OCR Job Template
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: Should this be sentence cased?

File: Documentation/BatchDocumentation/samples/ocr/readme.md [hunk 17]
```diff
@@ -1,36 +1,78 @@
...
 ## Prerequisites
+
 You must have an Azure Batch account set up with a linked Azure Storage account.
 
 ## Create a pool
-Run `azure batch pool create --template pool.json` to create your pool using the default settings (A pool named 'ocr-pool' with 3 STANDARD_D1_V2 VMs). 
 
-If you want to change the default values of the pool creation, you can create a JSON file to supply the parameters of your pool. If you have a large number of files 
-to convert, you should use a larger pool or bigger VMs in the pool. In order to create the pool with your own configurations, run `azure batch pool create --template pool.json --parameters <your settings JSON file>`.
+To create your pool using the default settings (A pool named 'ocr-pool' with 3 STANDARD_D1_V2 VMs):
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: "A" -> "a" and check pool configuration text boilerplate

File: Documentation/BatchDocumentation/samples/ocr/readme.md [hunk 18]
```diff
@@ -1,36 +1,78 @@
...
 ## Upload files
-Run command `azure batch file upload <path> <group>` on a folder containing PDF files which are named with numerically increasing names (i.e. `1.pdf`, `2.pdf`, `3.pdf`, etc).
+
+To upload your PDF files:
+
+```bash
+azure batch file upload <path> <group>
+```
+
+Run this command on a folder containing PDF files which are named with numerically increasing names (i.e. `1.pdf`, `2.pdf`, `3.pdf`, etc). It is important that there are no gaps in the numbering of the files.
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: If we have task-per-file now then we should fix this

File: Documentation/BatchDocumentation/samples/mpi/readme.md [hunk 19]
```diff
@@ -5,35 +5,81 @@ This template shows how to use `MS-MPI` to run MPI work.
...
-Run command `azure batch file upload <path> <group>` on a folder containing MPIHelloWorld.exe and its dependencies which are named with numerically increasing names with `sample` prefix (i.e. `sample1.wav`, `sample2.wave`, `sample3.wav`, etc).
 
-**Build a Release version of MPIHelloWorld.exe so that you don't have to include any additional dependencies as resource files (ex: msvcp140d.dll or vcruntime140d.dll).
+Upload files from a folder:
+
+```bash
+azure batch file upload <path> <group>
+```
+
+Run this in a folder containing MPIHelloWorld.exe and its dependencies. The parametric sweep expects the files to be named `sample1.wav`, `sample2.wav`, `sample3.wav` and so on - each with the prefix `sample` and an increasing index number. It's important that your files are sequentially numbered with no gaps.
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: Thought I saw task-per-file mentioned elsewhere - if we have that, why do we need the sequential numbering?

File: Documentation/BatchDocumentation/samples/ocr/readme.md [hunk 20]
```diff
@@ -1,36 +1,78 @@
...
 
-1. `poolId` must match the pool you created earlier.
-2. `firstPdfId` must match the first PDF file you uploaded earlier (specify `1` to reference `1.pdf`).
-3. `lastPdfId` must match the last PDF file you uploaded earlier (specify `10` to reference `10.pdf`).
-4. `inputFileGroup` must match the name of the group used in the `azure batch file upload` command earlier.
-5. `outputFileStorageUrl` must be a writable SAS to an Azure Storage container.
+
+| Parameter            | Required  | Description                                                                                                                                    |
+| -------------------- | --------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| jobId                | Mandatory | The id of the Azure Batch job. <br/> Must match the id of the pool you created earlier.                                                        |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: Job ID must match pool ID?  Are you sure this is correct?

File: Documentation/BatchDocumentation/samples/readme.md [hunk 21]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+| [Create Pool and Job](hello-world/create-pool-and-job)                               | Create a pool and then run a job with a single task. Both the pool and the job are defined using templates with hard coded values.                                                                                                                                                                                                            |
+| [Create Pool and Job with templates](hello-world/create-pool-and-job-with-templates) | Create a pool and then run a job with a single task. Both the pool and the job are defined using a parameterized templates. Parameter values used to fill out the templates are stored in separate files that are easy to modify as required.                                                                                                 |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: weirdly mixed casing.  Should be either "Create Pool and Job with Templates" or "Create pool and job with templates" surely?

File: Documentation/BatchDocumentation/samples/readme.md [hunk 22]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+| [Task Per File](hello-world/taskPerFile)                                             | Run a specific piece of processing independently across a set of files that are uploaded into storage. The job is specified as a template accepting parameters.                                                                                                                                                                               |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: inconsistent directory conventions: create-pool-and-job a la snake case vs. taskPerFile a la camel case

File: Documentation/BatchDocumentation/samples/readme.md [hunk 23]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+| [Task Per File with Application Template](hello-world/taskPerFile-appTemplate)       | Run a specific piece of processing independently across a set of files that are uploaded into storage. The job is split into two parts - a reusable *application template* defining the required processing and a specific use *job* that references the template while specifying parameters, pool information and other management details. |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: I get the reason for the 'hello world' grouping but it feels a bit intimidating opening the 'hello world' folder and having to choose between 4 different options and it's not clear how they relate or which to start with.  Can we tell a more incremental story here?

Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: There's a lack of orthogonality in the hello world samples - application templates are bundled in with task per file

Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: I found the sentence beginning 'the job is split into two parts' awkward and the clause 'a specific use *job*' hard to read.

File: Documentation/BatchDocumentation/samples/readme.md [hunk 24]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
-
-Blender is an open-source 3D content creation suite. This sample demonstrates distributed rendering on Azure Batch.
-
-### [Docker - Caffe](docker)
-
-Caffe is an open-source deep learning framework. This sample demonstrates configuration of Caffe via Docker integration using 
-Shipyard.
+| Sample                                                | Description                                                                                                                                                                                                          |
+| ----------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [FFmpeg](ffmpeg)                                      | FFmpeg is an open-source command line tool for processing multimedia files. This is a sample demonstrating audio compression with Azure Batch on a large number of numerically-named files using a parametric sweep. |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: You highlight the use of parametric sweep on this one but not on OCR?

File: Documentation/BatchDocumentation/samples/readme.md [hunk 25]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+| [OCR](ocr)                                            | OCR (Optical Character Recognition) is the process of extracting text from PDF images. This sample demonstrates the batch processing of PDF files.                                                                   |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: This is not helpful to a user of the sample.  Very few people will be coming to these samples asking "How do I do OCR on Batch" - you need to tell them what *Batch features* each sample demonstrates and why they might be relevant to the solution developer.

File: Documentation/BatchDocumentation/samples/readme.md [hunk 26]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+| [MPI](mpi)                                            | This sample demonstrates the batch run a MPI task with MultiInstanceSettings feature.                                                                                                                                |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: "demonstrates the batch run a MPI task" is flawed English.  Also is this about demonstrating the use of MPI/multiInstanceSettings, or demonstrating how to use MPI/multiInstanceSettings in a template?  Text above implied the latter but I could be misunderstanding.

File: Documentation/BatchDocumentation/samples/readme.md [hunk 27]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+| [Blender](blender)                                    | Blender is an open-source 3D content creation suite. This sample demonstrates distributed rendering on Azure Batch.                                                                                                  |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: Again, as a solution developer wanting to find how to implement *my* stuff, why would I look further at this sample?

File: Documentation/BatchDocumentation/samples/readme.md [hunk 28]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+| [Blender (Application Template)](blender-appTemplate) | A variation of the [Blender](blender) sample that uses an application template to separate job definiton and management.                                                                                            |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: "definiton" -> "definition"

File: Documentation/BatchDocumentation/samples/readme.md [hunk 29]
```diff
@@ -6,45 +6,22 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+| [Docker - Caffe](docker)                              | Caffe is an open-source deep learning framework. This sample demonstrates configuration of Caffe via Docker integration using Shipyard.                                                                              |
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: This is more helpful but Shipyard is left unexplained.  Is it relevant from the XPlat CLI point of view?  If so, needs link or explanation.  If not, just talk about 'Docker integration with the XPlat CLI' or something like that.

File: Documentation/BatchDocumentation/samples/blender/readme.md (hunk 6 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/blender/readme.md]: Fixed the title.

File: Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md (hunk 7 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/hello-world/create-pool-and-job-with-templates/readme.md]: I could only speculate; I've assumed that the original authors of the samples had reasons for their choices. 

**Should I change them?**

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 11 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: Reformatted to follow the practice used elsewhere of showing the details below the command line. Added blank lines following headings.

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 2 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: Changed to "... don't forget to delete *this pool* through the Azure portal ..."

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 12 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: I've asked @xingwu1 to comment.

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 19 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: I've asked @xingwu1 to comment.

File: Documentation/BatchDocumentation/samples/ocr/readme.md (hunk 18 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: Modified to use taskPerFile as suggested.

File: Documentation/BatchDocumentation/samples/mpi/readme.md (hunk 19 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/mpi/readme.md]: Going to leave this sample as a parametric sweep, purely for contrast.

File: Documentation/BatchDocumentation/samples/ocr/readme.md (hunk 20 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: Wrong table row. 🙁 Fixed.

File: Documentation/BatchDocumentation/samples/readme.md (hunk 23 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/samples/readme.md]: Reworded.

File: Documentation/BatchDocumentation/samples/readme.md [hunk 30]
```diff
@@ -6,22 +6,81 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
-| [Task Per File](hello-world/taskPerFile)                                             | Run a specific piece of processing independently across a set of files that are uploaded into storage. The job is specified as a template accepting parameters.                                                                                                                                                                               |
-| [Task Per File with Application Template](hello-world/taskPerFile-appTemplate)       | Run a specific piece of processing independently across a set of files that are uploaded into storage. The job is split into two parts - a reusable *application template* defining the required processing and a specific use *job* that references the template while specifying parameters, pool information and other management details. |
+### [Create pool and job](hello-world/create-pool-and-job)
 
-## Real World Samples
+Create a pool and then run a job with a single task. Both the pool and the job are defined using templates with hard coded values. 
+
+Features used:
+
+* [Pool and Job templates with parameterization](../templates.md)
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: lower-case j in job?

File: Documentation/BatchDocumentation/samples/readme.md [hunk 31]
```diff
@@ -6,22 +6,81 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+
+
+### [Create pool and job with templates](hello-world/create-pool-and-job-with-templates)
+
+Create a pool and then run a job with a single task. Both the pool and the job are defined using a parameterized templates. Parameter values used to fill out the templates are stored in separate files that are easy to modify as required.  
+
+Features used:
+
+* [Pool and Job templates with parameterization](../templates.md)
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: casing?

File: Documentation/BatchDocumentation/samples/readme.md [hunk 32]
```diff
@@ -6,22 +6,81 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+
+### [Task per file with application template](hello-world/task-per-file-with-app-template)
+
+Run a specific piece of processing independently across a set of files that are uploaded into storage. The actual processing involved is split out into a separate *application template*. The job itself references the template while specifying parameters, pool information and other management details. Application templates are intended to be flexible and reusable across a number of jobs.
+
+Features used:
+
+* [Split job configuration and management with reusable application templates](../application-templates.md)
+* [Task per file task factory](../taskFactories.md#task-per-file)
+* [Automatic task output-file persistence to Azure Storage](../outputFiles.md)
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: I found this title a bit hard to parse (the hyphen confused me I think). How about "Automatically persisting task output files to Azure Storage" or "Automatic persistence of task output files to Azure Storage"?

File: Documentation/BatchDocumentation/samples/readme.md [hunk 33]
```diff
@@ -6,22 +6,81 @@ Here are a collection of samples to demonstrate the new features in this Batch p
...
+
+FFmpeg is an open-source command line tool for processing multimedia files. This is a sample demonstrating audio compression with Azure Batch on a large number of numerically-named files using a parametric sweep.
+
+### [OCR](ocr)
+
+OCR (Optical Character Recognition) is the process of extracting text from PDF images. This sample demonstrates the batch processing of PDF files.
+
+## Features used:
+
+* [Support for job and pool templates with parameterization](../templates.md)
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/readme.md]: Elsewhere the 'Support for' was omitted and the order was 'pool and job' #pickypickypicky

File: Documentation/BatchDocumentation/application-templates.md [hunk 34]
```diff
@@ -35,11 +35,11 @@ This sample Blender job specifies rendering of a Blender scene using application
...
 The `applicationTemplateInfo` element gives the `filePath` to the application template and provides all the `parameters` required to configure the application template for use.
 
-## Samples
+### Samples
 
-[**Blender**](samples/blender-appTemplate/readme.md) - using an application template to specify the rendering of a Blender scene. This sample uses the experimental *[parametric sweep](taskFactories.md#parametric-sweep)* [task factory](taskFactories.md) feature.
+The following samples use application templates:
 
-[**Data Movement**](samples/Helloworld/data-movement-appTemplate/readme.md) - using an application template to specify processing of a number of input files, with the results of processing automatically uploaded into blob storage afterwards. This sample uses the experimental *[task per file](taskFactories.md#task-per-file)* [task factory](taskFactories.md) feature.
+* [Task Per File](samples/hello-world/task-per-file)
```
Comment (itowlson) [inline Documentation/BatchDocumentation/application-templates.md]: Is that now the only one?  The Blender app template sample got removed?

File: Documentation/BatchDocumentation/readme.md [hunk 35]
```diff
@@ -2,14 +2,14 @@
...
-* [Input data upload to Batch linked storage accounts](Documentation/BatchDocumentation/inputFiles.md#input-file-upload)
-* [Input data references using linked storage accounts](Documentation/BatchDocumentation/inputFiles.md#referencing-input-data)
-* [Automatic task output-file persistence to Azure Storage](Documentation/BatchDocumentation/outputFiles.md)
-* [Support for job and pool templates with parameterization](Documentation/BatchDocumentation/templates.md)
-* [Task factories for automatic task generation on job submission](Documentation/BatchDocumentation/taskFactories.md)
-* [Container-based workflows through integration with Batch Shipyard (Docker)](Documentation/BatchDocumentation/shipyard.md)
-* [Split job configuration and management with reusable application templates](Documentation/BatchDocumentation/application-templates.md)
-* [Package manager support for easy software installation](Documentation/BatchDocumentation/packages.md)
+* [Input data upload to Batch linked storage accounts](inputFiles.md#input-file-upload)
+* [Input data references using linked storage accounts](inputFiles.md#referencing-input-data)
```
Comment (itowlson) [inline Documentation/BatchDocumentation/readme.md]: I feel like these two features kind of want to be explained together - it's basically "getting your input data into a task without having to deal with SASes" but I'm not sure how to word that!

File: Documentation/BatchDocumentation/samples/ocr/job.json [hunk 36]
```diff
@@ -47,21 +35,17 @@
...
+                },
                 "repeatTask": {
-                    "commandLine": "gs -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -sDEVICE=pngalpha -sOutputFile={0}-%03d.png -r250 {0}.pdf && for f in *.png; do tesseract $f ${{f%.*}}; done",
+                    "commandLine": "gs -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -sDEVICE=pngalpha -sOutputFile={fileNameWithoutExtension}-%03d.png -r250 {fileNameWithoutExtension}.pdf && for f in *.png; do tesseract $f ${{f%.*}}; done",
                     "resourceFiles": [
                         {
                             "source": {
                                 "fileGroup": "[parameters('inputFileGroup')]",
-                                "prefix": "{0}.pdf"
+                                "prefix": "{fileNameWithoutExtension}.pdf"
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/ocr/job.json]: perhaps this should be {fileName} - otherwise if the container contains a file with a non-.pdf extension it will look for the wrong resource file and give a confusing error - guess the same applies to the input file name on the gs command line

File: Documentation/BatchDocumentation/samples/ocr/readme.md [hunk 37]
```diff
@@ -1,36 +1,85 @@
...
 ## Upload files
-Run command `azure batch file upload <path> <group>` on a folder containing PDF files which are named with numerically increasing names (i.e. `1.pdf`, `2.pdf`, `3.pdf`, etc).
+
+To upload your PDF files:
+
+```bash
+azure batch file upload <path> <group>
+```
+
+Run this command on a folder containing PDF files which are named with numerically increasing names (i.e. `1.pdf`, `2.pdf`, `3.pdf`, etc). It is important that there are no gaps in the numbering of the files.
```
Comment (itowlson) [inline Documentation/BatchDocumentation/samples/ocr/readme.md]: can probably the stuff about numbering and gaps now?

File: lib/commands/batch/batch.jobUtils._js (hunk 5 above)
Comment (xingwu1) [inline lib/commands/batch/batch.jobUtils._js]: In batch-beta branch, the whole jobUtil file is using 4 spaces for intent (don't know why it is different compare to other files). It would be great if you can update it.

File: Documentation/BatchDocumentation/readme.md (hunk 35 above)
Comment (theunrepentantgeek) [inline Documentation/BatchDocumentation/readme.md]: Yeah, I'm not sure how to word them either. I'm also blocked by thinking about them from an implementation perspective - one refers to the new `azure batch file upload` command, the other to the `taskPerFile` task factory.
